views.go

```diff
@@ -201,43 +233,45 @@ func (lk *logKeeper) createTest(w http.ResponseWriter, r *http.Request) {
 		lk.render.WriteJSON(w, http.StatusNotFound, apiError{Err: "build not found"})
 		return
 	}
+	if build.S3 {
+		exists, err := lk.opts.Bucket.CheckMetadata(r.Context(), buildID, "")
+		if err != nil {
+			lk.logErrorf(r, "checking metadata in build '%s': %v", buildID, err)
+			lk.render.WriteJSON(w, http.StatusInternalServerError, apiError{Err: "finding build"})
+			return
+		}
+		if !exists {
+			lk.render.WriteJSON(w, http.StatusInternalServerError, apiError{Err: "build not found"})
+			return
+		}
+	}
+	newTest.BuildName = build.Name
```

Comment:
Can't we just move the creation of newTest down one line so that we can continue to use the object literal syntax to set BuildName?

Reply:
This code is to be deleted once we remove the DB writes, so I just reorganized it so it is simpler to remove it once we are ready. I am not opposed to moving it back though if you feel strongly.

Reply:
Ah, I see, because we plan on deleting buildname. That makes sense.